server/src/domain/system-config/dto/system-config-machine-learning.dto.ts

```diff
@@ -1,4 +1,74 @@
-import { IsBoolean, IsUrl, ValidateIf } from 'class-validator';
+import { ApiProperty } from '@nestjs/swagger';
+import { Type } from 'class-transformer';
+import {
+  IsBoolean,
+  IsEnum,
+  IsNumber,
+  IsObject,
+  IsOptional,
+  IsString,
+  IsUrl,
+  Max,
+  Min,
+  ValidateIf,
+  ValidateNested,
+} from 'class-validator';
+
+export enum ModelType {
+  IMAGE_CLASSIFICATION = 'image-classification',
+  FACIAL_RECOGNITION = 'facial-recognition',
+  CLIP = 'clip',
+}
+
+export enum CLIPMode {
+  VISION = 'vision',
+  TEXT = 'text',
+}
+
+export class ModelConfig {
+  @IsBoolean()
+  enabled!: boolean;
+
+  @IsString()
+  @ApiProperty({ type: 'string' })
+  modelName!: string;
+
+  @IsEnum(ModelType)
+  @ApiProperty({ type: 'string', enum: ModelType, enumName: 'ModelType' })
```

Comment:
Does this even need to be passed in and saved?

Reply:
Not really. It's just needed for the request itself, so I can add it in the model-specific handlers in the ML repository.

Reply:
Yeah, I would probably remove it everywhere and send it in the infra/repository implementation.